CRM/Sqltasks/Task.php

```diff
@@ -263,6 +267,10 @@ public function execute($params = []) {
     // 2. run the actions
     $actions = CRM_Sqltasks_Action::getAllActiveActions($this);
     foreach ($actions as $action) {
+      $action_table = $action->getConfigValue('table');
+      $this->resolveTokens($action_table, $random_value, 'random');
+      $action->setConfigValue('table', $action_table);
```

Comment:
This does not work for `CRM_Sqltasks_Action_SyncGroup` or `CRM_Sqltasks_Action_SyncTag`, where the table is stored in `contact_table`. Stylistically, it feels like replacing the tokens where needed is a concern for the actions themselves, rather than `CRM_Sqltasks_Task` (which could just pass the random value to the actions) - we want some level of separation there for extendability.

Reply:
"This does not work for CRM_Sqltasks_Action_SyncGroup or CRM_Sqltasks_Action_SyncTag, where the table is stored in contact_table" - this i understand.
Could you clarify how this process should work like?

Reply:
> Could you clarify how this process should work like?

A generalized approach (that could be reused for other things in the future) might look like this:

```php
abstract class CRM_Sqltasks_Action {

  // ...
  protected $context = [];

  /**
   * Set execution context
   *
   * @param array $context
   */
  public function setContext(array $context) {
    $this->context = $context;
  }

  // ...
}

// in the loop in CRM_Sqltasks_Task::execute():
$action->setContext(['random' => $random_value]);

// in actions that have table parameters, e.g. CRM_Sqltasks_Action_ContactSet
/**
 * get the table with the contact_id column
 */
public function getContactTable() {
  $table_name = $this->getConfigValue('contact_table');
  // TODO: replace token in $table_name with $this->context['random'];
  return trim($table_name);
}
```